File: Data Structures & Algorithms/binary-tree-maximum-path-sum/submission-1.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */

class Solution {
    int fun(TreeNode* node,int& maxi){
        if(!node)return 0;
        int l=max(fun(node->left,maxi),0);
        int r=max(fun(node->right,maxi),0);
        maxi=max(maxi,l+r+node->val);
        return max(l,r)+node->val;
    }
public:
    int maxPathSum(TreeNode* root) {
        int maxi=root->val;
         fun(root,maxi);
         return maxi;
    }
};

File: Data Structures & Algorithms/binary-tree-right-side-view/submission-1.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def rightSideView(self, root: Optional[TreeNode]) -> List[int]:
        if not root:
            return []

        mp = {}
        q = deque()
        q.append((0, root))

        while q:
            level, node = q.popleft()

            if level not in mp:
                mp[level] = node.val

            if node.right:
                q.append((level + 1, node.right))
            if node.left:
                q.append((level + 1, node.left))

        ans = []
        for level in sorted(mp.keys()):
            ans.append(mp[level])

        return ans

